template.go

```diff
@@ -160,11 +166,61 @@ func (cmd *TemplateCommand) executeTemplateSecrets(client *Client, input string)
 
 		// For each of the secret keys, lookup the value
 		for k, placeholder := range kv {
-			if v, ok := secret.Data[k].(string); ok {
-				content = strings.Replace(content, placeholder, v, -1)
-			} else {
-				return "", fmt.Errorf("secret %s: key %q not found", path, k)
+			if strings.HasPrefix(placeholder, "_VAULT_STRING_") {
+				if v, ok := secret.Data[k].(string); ok {
+					content = strings.Replace(content, placeholder, v, -1)
+				} else {
+					return "", fmt.Errorf("secret %s: key %q not found", path, k)
+				}
+			}
+		}
+	}
+
+	return
+}
+
+func (cmd *TemplateCommand) executeTemplateNested(client *Client, input string) (content string, err error) {
+	content = input
+
+	// For each of the secret paths, lookup the secret
+	for path, kv := range cmd.lookup {
+		var secret *api.Secret
+		if secret, err = client.Logical().Read(strings.TrimLeft(path, "/")); err != nil {
+			return
+		}
+		if secret == nil {
+			return "", fmt.Errorf("nested %s: not found", path)
+		}
+
+		// For each of the secret keys, lookup the value
+		for k, placeholder := range kv {
+			if strings.HasPrefix(placeholder, "_VAULT_NESTED_") {
+				keys := strings.Split(k, ".")
+
+				var nestedData map[string]interface{}
+				if err := json.Unmarshal([]byte(secret.Data[keys[0]].(string)), &nestedData); err != nil {
+					return "", fmt.Errorf("nested %s/%s: failed to parse JSON", path, keys[0])
+				}
+
+				mydata := nestedData
+				levels := len(keys) - 1
+				for _, nestedkey := range keys[1:] {
+					if levels > 1 {
+						if mydata[nestedkey] == nil {
+							return "", fmt.Errorf("nested %s: key %q not found", path, nestedkey)
+						}
+						mydata = mydata[nestedkey].(map[string]interface{})
+					} else {
+						if reflect.TypeOf(mydata[nestedkey]).Kind() == reflect.String {
```

Comment:
is `reflect` needed here? or some type assertion like 
```go
_, ok = mydata[nestedkey].(string)
```
is enough?

Reply:
@luisyonaldo type assertion should do.